src/firmware/app/control/trajectory_planner.h

```diff
@@ -0,0 +1,273 @@
+ * interpolation period equivalent of the input trajectory
+ * @param variable_time_trajectory [in] The valid trajectory used as
+ * reference for a constant interpolation period trajectory
+ * @param interpolation_period [in] The constant change in time that corresponds to each
+ * trajectory segment
+ *
+ */
+void app_trajectory_planner_interpolateConstantTimeTrajectorySegmentation(
+    Trajectory_t constant_period_trajectory[TRAJECTORY_PLANNER_MAX_NUM_ELEMENTS],
+    Trajectory_t variable_time_trajectory[TRAJECTORY_PLANNER_MAX_NUM_ELEMENTS],
+    const float interpolation_period);
+
+
+/**
+ * Returns a the total length of any arc specified by an arc parameterizaton
+ *
+ * @param arc_length_param [in] Arc length parameterization
+ *
+ * @return [out] The arc length in meters
+ */
+static float app_trajectory_planner_getTotalArcLength(
+    ArcLengthParametrization_t arc_length_parameterization);
+
+/**
+ * Generates the X/Y points for each position on the constant arc length trajectory
+ *
+ * @pre traj_elements is pre-allocated up the the limit specified by
+ * TRAJECTORY_PLANNER_MAX_NUM_ELEMENTS
+ *
+ * @param path [in] The polynomial that defines the path of the trajectory
+ *
+ * @param num_elements [in] The number of elements in the trajectory
+ *
+ * @param arc_length_parameterization [in] The arc length parameterization of the path
+ * polynomial
+ *
+ * @param arc_segment_length [in] The length of each segment in the trajectory
+ *
+ */
+static void app_trajectory_planner_generateConstArclengthTrajectoryPositions(
+    TrajectoryElement_t traj_elements[TRAJECTORY_PLANNER_MAX_NUM_ELEMENTS],
+    Polynomial2dOrder3_t path, const unsigned int num_elements,
+    ArcLengthParametrization_t arc_length_parameterization,
+    const float arc_segment_length);
+
+/**
+ * Returns the maximum velocity profile for a given curve based on curvature and maximum
+ * allowable acceleration
+ *
+ * @pre max_allowable_speed_profile is pre-allocated up the the limit specified by
+ * TRAJECTORY_PLANNER_MAX_NUM_ELEMENTS
+ *
+ * @param max_allowable_speed_profile [out] The pre-allocated array that will contain all
+ * of the maximum allowable velocity values for each point on the curve
+ *
+ * @param path [in] The polynomial that defines the path of the trajectory
+ *
+ * @param num_elements [in] The number of elements in the trajectory
+ *
+ * @param arc_length_parameterization [in] The arc length parameterization of the path
+ * polynomial
+ *
+ * @param arc_segment_length [in] The length of each segment in the trajectory
+ *
+ * @param max_allowable_acceleration [in] The max allowable acceleration at any discrete
+ * point on the trajectory
+ *
+ */
+static void app_trajectory_planner_getMaxAllowableSpeedProfile(
+    float max_allowable_speed_profile[TRAJECTORY_PLANNER_MAX_NUM_ELEMENTS],
+    Polynomial2dOrder3_t path, unsigned int num_elements,
+    ArcLengthParametrization_t arc_length_parameterization, float arc_segment_length,
```

Comment:
nit: should we make `arc_segment_length` const?

Reply:
Made const